src/WaveTrack.cpp

```diff
@@ -1457,21 +1461,16 @@ void WaveTrack::PasteWaveTrack(double t0, const WaveTrack* other)
     // Make room for the pasted data
     if (editClipCanMove) {
         if (!singleClipMode) {
-            // We need to insert multiple clips, so split the current clip and
-            // move everything to the right, then try to paste again
-            if (!IsEmpty(t0, GetEndTime())) {
-                auto tmp = Cut(t0, GetEndTime() + 1.0 / mRate);
-                Paste(t0 + insertDuration, tmp.get());
-            }
+            // We need to insert multiple clips, so split the current clip and ...
+            SplitAt(t0);
         }
-        else {
-            // We only need to insert one single clip, so just move all clips
-            // to the right of the paste point out of the way
-            for (const auto& clip : mClips)
-            {
-                if (clip->GetPlayStartTime() > t0 - (1.0 / mRate))
-                    clip->Offset(insertDuration);
-            }
+        //else if there is a clip at t0 insert new clip inside it and ...
+       
+        // ... move everything to the right
+        for (const auto& clip : mClips)
+        {
+            if (clip->GetPlayStartTime() > t0 - (1.0 / mRate))
```

Comment:
I could ask the same here.

But, I see this only preserves old logic, whether that treats the boundary condition right or wrong.

Maybe there is a bug to discover here, but not one introduced by this commit.